automation-roles/30-provision-infra/destroy-aws/tasks/destroy-aws-ipi.yml

```diff
@@ -10,6 +10,41 @@
   vars:
     _p_ocp_version: "{{ _current_openshift_cluster.ocp_version }}"
 
+- name: Check if metadata.json exists
+  stat: 
+    path: "{{ status_dir }}/aws-ipi/{{ _current_openshift_cluster.name }}/metadata.json"
+  register: _openshift_metadata_json
+
+- block:
+  - name: Retrieve metadata json from the vault
+    include_role: 
+      name: vault-get-secret
+    vars:
+      secret_name: "{{ environment_name }}-metadata-json"
```

Comment:
The secret must  be linked to the openshift cluster name, not the environment name. The environment name is the key we use to group secrets and nothing else.